erpnext/public/js/controllers/transaction.js

```diff
@@ -1856,18 +1856,16 @@ erpnext.TransactionController = class TransactionController extends erpnext.taxe
 
 	apply_rule_on_other_items(args) {
 		const me = this;
-		const fields = ["discount_percentage", "pricing_rules", "discount_amount", "rate"];
+		const fields = ["pricing_rules"];
 
 		for(var k in args) {
 			let data = args[k];
 
 			if (data && data.apply_rule_on_other_items && JSON.parse(data.apply_rule_on_other_items)) {
+				fields.push(frappe.scrub(data.pricing_rule_for))
 				me.frm.doc.items.forEach(d => {
-					if (in_list(JSON.parse(data.apply_rule_on_other_items), d[data.apply_rule_on]) && d.item_code === data.item_code) {
+					if (JSON.parse(data.apply_rule_on_other_items).includes(d[data.apply_rule_on])) {
 						for(var k in data) {
-							if (data.pricing_rule_for == "Discount Percentage" && data.apply_rule_on_other_items && k == "discount_amount") {
-								continue;
-							}
```

Comment:
This were added to fix this scenario https://github.com/frappe/erpnext/pull/41956

Can you check if they still work

Reply:
Yes, it will work.

The code above #41956  works effectively. When pricing_rule_for is set to "Discount Percentage," the field (k) must not be "discount_amount," 

but 
there is no need to update the rate in this case.right?

In scenarios where pricing_rule_for is "Rate," the field (k) should exclude both discount_amount and discount_percentage.

To address all these cases, I updated the variable(fields) to dynamically adjust themselves as required.

This solution ensures that all scenarios under rate_or_discount are thoroughly managed, considering "Discount Percentage," "Rate," and "Discount Amount," as applied to other items.